spring-boot-consumer/src/main/java/com/solace/samples/spring/boot/consumer/SolaceGuaranteedConsumer.java

```diff
@@ -0,0 +1,117 @@
+import com.solace.messaging.config.MissingResourcesCreationConfiguration;
+import com.solace.messaging.config.SolaceProperties;
+import com.solace.messaging.config.profile.ConfigurationProfile;
+import com.solace.messaging.receiver.DirectMessageReceiver;
+import com.solace.messaging.receiver.MessageReceiver;
+import com.solace.messaging.receiver.PersistentMessageReceiver;
+import com.solace.messaging.resources.Queue;
+import com.solace.messaging.resources.TopicSubscription;
+import com.solace.samples.spring.boot.config.SolaceConfigProperties;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PreDestroy;
+import java.util.Properties;
+
+@Component
+@Slf4j
+public class SolaceGuaranteedConsumer {
+
+    @Autowired
+    private SolaceConfigProperties configProperties;
+    private PersistentMessageReceiver persistentMessageReceiver;
+    private MessagingService messagingService;
+
+    @EventListener
+    public void onApplicationEvent(final ApplicationReadyEvent applicationReadyEvent) throws InterruptedException {
+
+        //1. Set up the properties including username, password, vpnHostUrl and other control parameters.
+        final Properties properties = setupPropertiesForConnection();
+
+        //2. Create the MessagingService object and establishes the connection with the Solace event broker
+        messagingService = MessagingService.builder(ConfigurationProfile.V1).fromProperties(properties).build();
+        messagingService.connect();  // This is a blocking connect action
+        setupConnectivityHandlingInMessagingService(messagingService);
+
+        //3. Build and start the receiver object
+        persistentMessageReceiver = messagingService.createPersistentMessageReceiverBuilder()
+                .withRequiredMessageClientOutcomeOperationSupport(
+                        new MessageAcknowledgementConfiguration.Outcome[]{Outcome.ACCEPTED, Outcome.FAILED, Outcome.REJECTED})
+                .withMissingResourcesCreationStrategy(                      // Configures the missing resources creation strategy.
+                        MissingResourcesCreationConfiguration.MissingResourcesCreationStrategy.CREATE_ON_START)   // The strategy to attempt create missing resources when the connection is established.
+                //.withMessageAutoAcknowledgement()                         // Client message ack is default behavior.  Enable for auto ack
+
+                .build(Queue.durableExclusiveQueue(configProperties.getQueueName()))           // If it does not exist, this configuration will provision the non-durable queue on the broker when the start() method is called.
```

Comment:
Can you please put in a comment that other queue types can also be created as required ? Link to the Queue object: // It is also possible to create different types of Queues as required : https://docs.solace.com/API-Developer-Online-Ref-Documentation/pubsubplus-java/com/solace/messaging/resources/Queue.html